packages/onchainkit/src/identity/hooks/useAddresses.ts

```diff
@@ -0,0 +1,31 @@
+import { getAddresses } from '@/identity/utils/getAddresses';
+import { DEFAULT_QUERY_OPTIONS } from '@/internal/constants';
+import { useQuery } from '@tanstack/react-query';
+import { mainnet } from 'viem/chains';
+import type {
+  GetAddressReturnType,
+  UseAddressesOptions,
+  UseQueryOptions,
+} from '../types';
+
+/**
+ * A React hook that leverages the `@tanstack/react-query` for fetching and optionally caching
+ * multiple Ethereum addresses from ENS names or Basenames in a single batch request.
+ */
+export const useAddresses = (
+  { names, chain = mainnet }: UseAddressesOptions,
+  queryOptions?: UseQueryOptions<GetAddressReturnType[]>,
+) => {
+  const namesKey = names.join(',');
+  const queryKey = ['useAddresses', namesKey, chain.id];
+
+  return useQuery<GetAddressReturnType[]>({
+    queryKey,
+    queryFn: () => getAddresses({ names, chain }),
+    enabled: !!names.length,
+    ...DEFAULT_QUERY_OPTIONS,
```

Comment:
nit: I'd recommend always putting defaults at the top of the object, otherwise if someone adds a default that conflicts with the keys above (unlikely in this case), things won't work as expected.

E.g. if someone added `enabled: true` to the default options, it'll take precedence over `enabled: !!names.length`